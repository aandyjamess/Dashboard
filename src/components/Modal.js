import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

import { mergeAll, path, keys } from "ramda";

import * as copy from "./../copy";

const Modal = ({actions, active, receipt}) => receipt && active >= 0 ? (
  <div className="modal">
    <span className="icon icon-dissmiss-mark" />
    <div className="modal-content">
      <span className="modal-close icon icon-dismiss-dark" onClick={actions.closeModal()} />

      {{
        share: "",
        application: (
          <div
            className="modal-header"
            style={{backgroundColor: path(["application", "appearance", "bg-color"])(receipt)}}
          >
            <img className="modal-header-logo" src={path(["application", "appearance", "bg-img"])(receipt)} />
          </div>
        )
      }[receipt.type]}

      <div className="modal-body modal-info">
        <span className="modal-avatar icon icon-user_selfie_ph_medium" />
        <span className="modal-check icon icon-receipt_confirmed" />
        <h4 className="modal-source"> {path(["application", "name"])(receipt)} </h4>
        <p className="modal-message">

          {{
            share: "",
            application: copy.message,
          }[receipt.type]}

        </p>
        <p className="modal-date">
          { moment(path(["transaction", "unix-timestamp"])(receipt) * 1000).format(copy.dateFormat.modal) }
        </p>
      </div>
      <div className="modal-footer">

        {(() => {
          const merged = mergeAll(path(["transaction", "attributes"])(receipt));

          return keys(merged).map(key => (
             <div key={key} className="modal-info-field">
               <small className="text-capitalize"> {key.split("-").join(" ")} </small><br />
               <span> {merged[key]} </span>
             </div>
          ))
        })()}

      </div>
    </div>
  </div>
) : <div />;

Modal.PropTypes = {
  active: PropTypes.bool,
  receipt: PropTypes.array
};

Modal.DefaultProps = {
  active: false,
  receipt: []
};

export default Modal;
