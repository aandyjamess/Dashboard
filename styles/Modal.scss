@import "./variables";

.modal {
  display: block;
  overflow: auto;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: $backdrop-color;
  z-index: 5000;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  &-content {
    width: 400px;
  }

  &-info {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &-close {
    cursor: pointer;
    margin-left: 16px;
    position: absolute;
    left: 100%;
  }

  &-avatar, &-check, &-source {
    margin-bottom: 24px;
  }

  &-message, &-date {
    margin-bottom: 12px;
  }

  &-header {
    border-radius: 5px 0;
    background-color: $color-7c;
    height: 140px;
    display: flex;
    align-items: center;
    justify-content: center;

    &-logo {
      width: 80px;
    }
  }

  &-footer {
    text-align: left;
  }

  &-info-field {
    margin-bottom: 20px;
  }
}
