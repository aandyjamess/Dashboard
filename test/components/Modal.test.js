import React from "react";
import toJson from "enzyme-to-json";

import { shallow } from "enzyme";

import Modal from "../../src/components/Modal";

describe("Modal", () => {
  it("renders correctly", () => {
    const actions = {openModal: () => {}, closeModal: () => {}};
    const wrapper = shallow(<Modal actions={actions} active={false} receipt={{}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});


