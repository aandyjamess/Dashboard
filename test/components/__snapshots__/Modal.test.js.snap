// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
<div
  className="modal"
>
  <span
    className="icon icon-dissmiss-mark"
  />
  <div
    className="modal-content"
  >
    <span
      className="modal-close icon icon-dismiss-dark"
    />
    <div
      className="modal-body modal-info"
    >
      <span
        className="modal-avatar icon icon-user_selfie_ph_medium"
      />
      <span
        className="modal-check icon icon-receipt_confirmed"
      />
      <h4
        className="modal-source"
      >
         
         
      </h4>
      <p
        className="modal-message"
      />
      <p
        className="modal-date"
      >
        Invalid date
      </p>
    </div>
    <div
      className="modal-footer"
    />
  </div>
</div>
`;
